extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -642,16 +646,14 @@ private static <DATA_TYPE> void encodePlain(@NotNull final ParquetInstructions w
                                 .asIntChunk();
                         lenChunk.copyToTypedBuffer(0, repeatCount, 0, lenChunk.size());
                         repeatCount.limit(lenChunk.size());
-                        transferObject.bufferFetchedData();
-                        // Return value ignored, this can lead to errors in case we are not able to fit all the data in
-                        // a single page, which is possible for vector<string> columns.
-                        columnWriter.addVectorPage(transferObject.getBuffer(), repeatCount, transferObject.rowCount());
+                        int numValuesBuffered = transferObject.bufferAllFetchedData();
```

Comment:
I think I would leave a comment here explaining why you must grab the entire vector (because it must be written in a single page, etc.) and there is a risk of OOM exceptions.

Reply:
I didn't paginate vector data because the implementation was becoming unnecessarily complicated in cases where we might run out of our page limit in the middle of the vector. 

- One approach is to pull out all elements of the vector, find the length in binary and then add it up to check if we are hitting the limit and only then transfer to buffer.
- Other way is to consider the limit in a relaxed manner and always put all elements of a single vector, and if we have hit or exceeded the limit, then stop.

Let me know if you want me to add this logic.

Reply:
There's complexity at the layer above, too.
See `io.deephaven.parquet.table.pagestore.topage.ToVectorPage`
and `io.deephaven.parquet.table.pagestore.topage.ToArrayPage`

Reply:
I think all we really want here is a comment that explains the problem (our code expecting all elements from an array or vector being on the same page, and a reference to that code) with a TODO referencing a new ticket that goes on the backlog.